specifyweb/frontend/js_src/lib/components/QueryBuilder/Line.tsx

```diff
@@ -167,7 +166,7 @@ export function QueryLine({
         canOpenMap = fieldName === 'latitude1' || fieldName === 'longitude1';
       } else if (isMapped)
         fieldType =
-          dataModelField?.isRelationship && relationshipIsToMany(dataModelField)
+          isFormatted && mappingPath.at(-1) === '#1'
```

Comment:
could you please apply the fixes I mentioned in https://github.com/specify/specify7/pull/4615#pullrequestreview-1932351672 in here?